flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-paimon/src/main/java/org/apache/flink/cdc/connectors/paimon/sink/PaimonMetadataApplier.java

```diff
@@ -129,25 +133,113 @@ private void applyCreateTable(CreateTableEvent event)
     private void applyAddColumn(AddColumnEvent event)
             throws Catalog.TableNotExistException, Catalog.ColumnAlreadyExistException,
                     Catalog.ColumnNotExistException {
-        List<SchemaChange> tableChangeList = new ArrayList<>();
-        event.getAddedColumns()
-                .forEach(
-                        (column) -> {
-                            SchemaChange tableChange =
-                                    SchemaChange.addColumn(
-                                            column.getAddColumn().getName(),
-                                            LogicalTypeConversion.toDataType(
-                                                    DataTypeUtils.toFlinkDataType(
-                                                                    column.getAddColumn().getType())
-                                                            .getLogicalType()));
-                            tableChangeList.add(tableChange);
-                        });
+        List<SchemaChange> tableChangeList = applyAddColumnEventWithPosition(event);
         catalog.alterTable(
                 new Identifier(event.tableId().getSchemaName(), event.tableId().getTableName()),
                 tableChangeList,
                 true);
     }
 
+    private List<SchemaChange> applyAddColumnEventWithPosition(AddColumnEvent event)
+            throws Catalog.TableNotExistException {
+        List<SchemaChange> tableChangeList = new ArrayList<>();
+        for (AddColumnEvent.ColumnWithPosition columnWithPosition : event.getAddedColumns()) {
+            SchemaChange tableChange;
+            switch (columnWithPosition.getPosition()) {
+                case FIRST:
+                    tableChange =
+                            SchemaChange.addColumn(
```

Comment:
What about declaring `applyAddColumnWithFirstPosition` and `applyAddColumnWithAfterPosition` too for consistency? And maybe we can extract some sharing logic (like column name, data conversion, and comments) to reduce code duplication.